### ISSUE TITLE
TypeError: cannot unpack non-iterable NoneType object (again)

### ISSUE THREAD

**Issue by nnobelis** [nexB/python-inspector#77]
This issue looks like a previous one (https://github.com/nexB/python-inspector/issues/72) but I get it on another file and the suggested fix does not work.
Therefore, I open a new issue.

While running `python-inspector --python-version 310 --json-pdt - --analyze-setup-py-insecurely --requirement requirements.txt`
on a file containing only:
```
non-existing-module==0.1
```
I get the following exception:
```
Traceback (most recent call last):
  File "/home/linuxbrew/.linuxbrew/bin/python-inspector", line 8, in <module>
    sys.exit(resolve_dependencies())
  File "/home/linuxbrew/.linuxbrew/opt/python@3.10/lib/python3.10/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/home/linuxbrew/.linuxbrew/opt/python@3.10/lib/python3.10/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/home/linuxbrew/.linuxbrew/opt/python@3.10/lib/python3.10/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/linuxbrew/.linuxbrew/opt/python@3.10/lib/python3.10/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/home/linuxbrew/.linuxbrew/opt/python@3.10/lib/python3.10/site-packages/click/decorators.py", line 26, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/home/linuxbrew/.linuxbrew/opt/python@3.10/lib/python3.10/site-packages/python_inspector/resolve_cli.py", line 389, in resolve_dependencies
    resolved_dependencies, purls = resolve(
  File "/home/linuxbrew/.linuxbrew/opt/python@3.10/lib/python3.10/site-packages/python_inspector/resolve_cli.py", line 472, in resolve
    resolved_dependencies, packages = get_resolved_dependencies(
TypeError: cannot unpack non-iterable NoneType object
```

Could the output of the tool be improved with `module 'non-existing-module' not found` for instance ?

Tested with python-inspector 0.7.1 and 0.8.1.

**Comment by nnobelis** [comment]
Please not that with python-inspector 0.6.4 the error message is different:

 ```
Resolving dependencies...
Using netrc file /home/`XXX/.netrc
direct_dependencies:
 DependentPackage(purl='pkg:pypi/non-existing-module@0.1', extracted_requirement='non-existing-module==0.1', scope='install')
environment: Environment(python_version='310', operating_system='linux')
repos:
 PypiSimpleRepository(index_url='https://pypi.org/simple', credentials=None)
ResolutionImpossible([RequirementInformation(requirement=<Requirement('non-existing-module==0.1')>, parent=None)])
nulldone!

``

**Comment by pombredanne** [comment]
I think that the error for non-existing package is not the same issue as the bug you reported in #78
We should handle gracefully non-existing packages with a proper message so I am reopening this.

**Comment by TG1999** [comment]
Fixed in version 0.9.0 here https://github.com/nexB/python-inspector/tree/v0.9.0
